### PR TITLE
Reduce use of UncheckedKeyHashMap in WebCore

### PR DESCRIPTION
**Pull request by cdumez**
#### 41e87066796a3c165d3f80e2286c756f999fb0c9
<pre>
Reduce use of UncheckedKeyHashMap in WebCore
<a href="https://bugs.webkit.org/show_bug.cgi?id=282642">https://bugs.webkit.org/show_bug.cgi?id=282642</a>

Reviewed by Ryosuke Niwa.

Reduce use of UncheckedKeyHashMap in WebCore, in places that are not
performance sensitive, for safety.

* Source/WebCore/loader/CrossOriginPreflightResultCache.h:
* Source/WebCore/loader/DocumentLoader.h:
* Source/WebCore/loader/ProgressTracker.h:
* Source/WebCore/loader/ResourceLoadStatistics.cpp:
* Source/WebCore/loader/appcache/ApplicationCache.h:
* Source/WebCore/loader/appcache/ApplicationCacheGroup.h:
* Source/WebCore/loader/appcache/ApplicationCacheStorage.h:
* Source/WebCore/loader/archive/ArchiveFactory.cpp:
* Source/WebCore/loader/archive/ArchiveResourceCollection.h:
* Source/WebCore/loader/cache/CachedImage.h:
* Source/WebCore/loader/cache/CachedResourceLoader.h:
* Source/WebCore/loader/cache/MemoryCache.h:
* Source/WebCore/platform/network/CredentialStorage.cpp:
(WebCore::CredentialStorage::findDefaultProtectionSpaceForURL):
* Source/WebCore/platform/network/CredentialStorage.h:
* Source/WebCore/platform/network/MIMEHeader.cpp:
* Source/WebCore/platform/network/ParsedContentType.h:
* Source/WebCore/platform/network/ResourceHandle.cpp:
* Source/WebCore/platform/network/cf/DNSResolveQueueCFNet.h:
* Source/WebCore/platform/network/cf/FormDataStreamCFNet.cpp:
(WebCore::httpBodyFromStream):
* Source/WebCore/platform/network/cocoa/RangeResponseGenerator.h:
* Source/WebCore/platform/network/cocoa/RangeResponseGenerator.mm:
(WebCore::RangeResponseGenerator::map):
(WebCore::RangeResponseGenerator::removeTask):
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::dumpStyleResolvers):
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/InternalsMapLike.h:
* Source/WebCore/testing/MockCDMFactory.h:
* Source/WebCore/testing/ServiceWorkerInternals.h:
* Source/WebCore/testing/WebFakeXRDevice.h:
* Source/WebCore/testing/WebFakeXRInputController.h:
* Source/WebCore/workers/service/server/SWRegistrationDatabase.cpp:
(WebCore::stripScriptSources):
(WebCore::populateScriptSourcesFromDisk):
(WebCore::SWRegistrationDatabase::importRegistrations):

Canonical link: <a href="https://commits.webkit.org/286207@main">https://commits.webkit.org/286207@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/d6fd78075aa12f0cfd289fc74e399013cec48a9a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/75125 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/54564 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/27965 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/79581 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/26381 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/77242 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/63704 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/2349 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/58972 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/17225 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/78192 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/49146 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/64536 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/39349 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/46514 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/22037 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/24705 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/67600 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/22377 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/81058 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/2455 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/1515 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/67229 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/2605 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/64541 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/66515 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/10459 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/8629 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/11600 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/2416 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/5220 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/2441 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/3370 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/2450 "Built successfully") | | |

### CODE REVIEW

**Review by zakariaridouh — DISMISSED** [review]
LGTM

(update: removing approval)

**Comment by rniwa** [comment]
Please don't approve a PR if you're not a reviewer yet.

**Review by rniwa — APPROVED** [review]

**Pull request merged**
